src/layout/List/ListComponent.tsx

```diff
@@ -35,108 +36,231 @@ export const ListComponent = ({ node }: IListProps) => {
-  );
-  const { data } = useDataListQuery(filter, dataListId, secure, mapping, queryParameters);
-  const calculatedDataList = (data && data.listItems) || defaultDataList;
+  const [sortColumn, setSortColumn] = useState<string | undefined>();
+  const [sortDirection, setSortDirection] = useState<AriaAttributes['aria-sort']>('none');
+
+  const filter: Filter = {
+    pageSize,
+    pageNumber,
+    sortColumn,
+    sortDirection,
+  };
 
-  const bindings = item.dataModelBindings || defaultBindings;
+  const { data } = useDataListQuery(filter, dataListId, secure, mapping, queryParameters);
+  const bindings = item.dataModelBindings ?? ({} as IDataModelBindingsForList);
   const { formData, setValues } = useDataModelBindings(bindings);
 
-  const handleChange = ({ selectedValue: selectedValue }: ChangeProps<Record<string, string>>) => {
-    const next: Record<string, string> = {};
+  const filteredHeaders = Object.fromEntries(Object.entries(tableHeaders).filter(([key]) => shouldIncludeColumn(key)));
+  const filteredRows: Row[] =
+    data?.listItems?.map((row) => {
+      const result = Object.fromEntries(Object.entries(row).filter(([key]) => shouldIncludeColumn(key)));
+      return result;
+    }) ?? [];
+
+  const selectedRow =
+    filteredRows.find((row) => Object.keys(formData).every((key) => row[key] === formData[key])) ?? '';
+
+  function handleRowSelect({ selectedValue }: { selectedValue: Row }) {
+    const next: Row = {};
     for (const binding of Object.keys(bindings)) {
       next[binding] = selectedValue[binding];
     }
     setValues(next);
-  };
+  }
 
-  const tableHeadersValues = { ...tableHeaders };
-  for (const key in tableHeaders) {
-    tableHeadersValues[key] = langAsString(tableHeaders[key]);
+  function shouldIncludeColumn(key: string): boolean {
+    return !isMobile || !tableHeadersMobile || tableHeadersMobile.includes(key);
   }
 
-  const selectedRow: Record<string, string> = React.useMemo(() => {
-    let matchRow: boolean[] = [];
-    if (!formData || Object.keys(formData).length === 0) {
-      return {};
-    }
-    for (const row of calculatedDataList) {
-      for (const key in formData) {
-        matchRow.push(formData[key] === row[key]);
-      }
-      if (!matchRow.includes(false)) {
-        return row;
-      }
-      matchRow = [];
-    }
-    return {};
-  }, [formData, calculatedDataList]);
+  function isRowSelected(row: Row): boolean {
+    return JSON.stringify(selectedRow) === JSON.stringify(row);
+  }
+
+  const title = item.textResourceBindings?.title;
+  const description = item.textResourceBindings?.description;
 
-  const renderPagination = () =>
-    pagination && (
+  if (isMobile) {
+    return (
+      <ComponentStructureWrapper node={node}>
+        <Radio.Group
+          role='radiogroup'
+          required={required}
+          legend={
+            <Heading
+              level={2}
+              size='sm'
+            >
+              {title}
```

Comment:
It seems we need to use the `<Lang />` feature here to get the value of the textresource.